pip/templates/deploy.py

```diff
@@ -30,48 +30,67 @@
 # noinspection PyUnresolvedReferences
 import twine.commands.upload
 
-pypi_profile = "{pypi_profile}"
-pip_registry = "{snapshot}" if "{snapshot}" else "{release}"
-
-if pypi_profile:
-    command = ['./dist/*', '--repository', pypi_profile]
-else:
-    pip_username, pip_password = (
-        os.getenv('DEPLOY_PIP_USERNAME'),
-        os.getenv('DEPLOY_PIP_PASSWORD'),
-    )
-
-    if not pip_username:
-        raise Exception(
-            'username should be passed via '
-            'DEPLOY_PIP_USERNAME env variable'
-        )
-
-    if not pip_password:
-        raise Exception(
-            'password should be passed via '
-            '$DEPLOY_PIP_PASSWORD env variable'
-        )
-    command = ['./dist/*', '-u', pip_username, '-p', pip_password, '--repository-url', pip_registry]
+dist_dir = "./dist"
+PYPRC_KEY = 'pyprc'
+SNAPSHOT_KEY = 'snapshot'
+RELEASE_KEY = 'release'
+
+ENV_DEPLOY_PIP_USERNAME = 'DEPLOY_PIP_USERNAME'
+ENV_DEPLOY_PIP_PASSWORD = 'DEPLOY_PIP_PASSWORD'
+
+repositories = {
+    PYPRC_KEY: "{pyprc_repository}",
+    SNAPSHOT_KEY: "{snapshot}",
+    RELEASE_KEY: "{release}"
+}
+
+parser = argparse.ArgumentParser()
+parser.add_argument('repo_type')
+
+def upload_command(repo_type_key, package_file, wheel_file):
```

Comment:
Packaged the command generation into a method that takes args about where to upload to and which files to upload